app/models/services/commitment_props.rb

```diff
@@ -0,0 +1,322 @@
+class Services::CommitmentProps
+  def initialize(commitment)
+    @commitment = commitment
+  end
+
+  def call
+    {
+      config:
+      {
+        action: @commitment.new_record? ? '/commitments.json' : "/commitments/#{ @commitments.id }.json",
+        method: @commitment.new_record? ? 'post' : 'put',
+        root_key: 'commitment'
+      },
+      survey: {
+        showPrevButton: false,
+        showQuestionNumbers: 'onPage',
+        showProgressBar: 'top',
+        requiredText: '',
+        pages: [
+          {
+            name: 'generalPage',
+            elements: [
+              {
+                type: 'text',
+                name: 'name',
+                title: 'Enter commitment name',
+                description:
+                  '(To save or continue with this form, you’ll need to provide a name for your commitment)',
+                isRequired: true
+              },
+              {
+                type: 'comment',
+                name: 'description',
+                title: 'Please provide a description for your commitment'
+              },
+              {
+                type: 'checkbox',
+                name: 'objectiveIds',
+                title: 'Please select the primary management objectives for the area',
+                description: 'Multi select option',
+                # defaultValue: [],
```

Comment:
is default value the same as 'selected'?

Reply:
No it isn't. I'm not sure that selected would work for all types of input so I just went with default value, it can take an array as well in surveyjs.